### DISCUSSION TITLE
Preview branches and environment variables

### DISCUSSION THREAD

**Discussion by dmessing** [vercel/community#1609] [Help] [answered]
### Summary

I have a single repo for my frontend with the following branches

- main --> prod
- staging --> staging
- develop --> development

Today the preview environment mirrors staging variables.
when i create a pull request and generate a preview right now it is grabbing the environment variables for the staging environment, i tried creating environment variables for preview (develop) branch but when issue a pull request say feature/abcd against development it is still taking the preview variables from staging.

Is it possible to have pull requests against develop use a  different set of environment variables?

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out, David. I'm assuming you set up the branch-specific environment variables like in this guide: [Adding Environment Variables](https://vercel.com/guides/set-up-a-staging-environment-on-vercel#adding-environment-variables). Please let me know if I misunderstood.

Preview deployments for PRs use the branch a change is coming from rather than the branch you are merging into. Once merged, the resulting deployment would then use applicable branch-specific environment variables. If we did it the other way around, you would end up with a bunch of preview deployments using production env vars for every PR into `main`. 

The most common setup is to have some default values for general preview deployments and override some or all of the them for deployments from the staging branch. Sort of the other way around from your current setup where it sounds like `staging` is your default preview environment. 

If that doesn't give you what you need for your workflow, the CLI and GitHub Actions can offer more flexibility to customize environment variables for branches or individual deployments. You can find more about that option in this guide: [How can I use GitHub Actions with Vercel?](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel)
